src/backend/cdb/cdbutil.c

```diff
@@ -957,8 +967,18 @@ cdbcomponent_recycleIdleQE(SegmentDatabaseDescriptor *segdbDesc, bool forceDestr
 			 lastWriter = cell, cell = lnext(cell)) ;
 
 		if (lastWriter)
+		{
+#ifdef FAULT_INJECTOR
+			if (SIMPLE_FAULT_INJECTOR("emulate_allocation_error") == FaultInjectorTypeSkip)
```

Comment:
That's a very general fault name. What about:
```suggestion
			if (SIMPLE_FAULT_INJECTOR("cdb_freelist_append_oom") == FaultInjectorTypeSkip)
```